app/helpers/CacheMiddleware.php

```diff
@@ -16,30 +19,33 @@
  *
  * @package helpers
  */
-class CacheMiddleware extends \Slim\Middleware
+class CacheMiddleware
 {
-    public function call()
-    {
-        /** @var $req \Slim\Http\Request */
-        $req = $this->app->request;
-        /** @var $res \Slim\Http\Response */
-        $res = $this->app->response;
+    public function __invoke(Request $req, Response $res, callable $next) {
 
         if ($this->shouldCache($req)) {
             $content = Cache::get($this->getCacheKey($req));
 
             if (!empty($content)) {
-                $res->setBody($content);
-
-                return;
+                if ($this->isJsonData($req)) {
+                    $res = $res->withHeader('Content-Type', 'application/json');
+                    $res->getBody()->write($content);
+                } else{
+                    $res->getBody()->write($content . "\n<!-- Cached response -->");
```

Comment:
is this comment fine to append to a JSON content type?

Reply:
I don't think so. That's why I'm only appending it when it isn't a json document.

Reply:
Stupid me :) didn't look at it properly